cupyx/scipy/sparse/_compressed.py

```diff
@@ -537,25 +537,23 @@ def _set_many(self, i, j, x):
         offsets = new_sp._get_arrayXarray(
             i, j, not_found_val=-1).astype(cupy.int32).ravel()
 
-        if -1 not in offsets:
+        mask = offsets > -1
+        self.data[offsets[mask]] = x[mask]
+
+        if mask.all():
```

Comment:
This is still doing device synchronization since you need to bring back the value to the host for the if comparisson, so I think that execution time won't benefit from these changes